Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -477,6 +478,21 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
             m_all_field_functors[lev][comp] = std::make_unique<DivBFunctor>(warpx.get_array_Bfield_aux(lev), lev, m_crse_ratio);
         } else if ( m_varnames[comp] == "divE" ){
             m_all_field_functors[lev][comp] = std::make_unique<DivEFunctor>(warpx.get_array_Efield_aux(lev), lev, m_crse_ratio);
+        } else if ( m_varnames[comp] == "sigma" ){
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(WarpX::em_solver_medium == MediumForEM::Macroscopic,
+                                             "sigma in plotfiles only works with algo.em_solver_medium=macroscopic");
```

Comment:
Thank you for adding in the suggestions Andy. Only one minor thing. I think this should ideally go in Diagnostics.cpp so that the abort is triggered as soon as the diagnostics are read in, along with other abort statements.

Reply:
Fixed in ba6ceaea2ddbd3a07374d96c2bad5d082d29e542